### PR TITLE
feat: support ignore user field while creating idempotency key

### PR DESCRIPTION
**Pull request by leedan77**
[why]
  * there may be different user model case, ignore user while creating
    key to support it

[how]
  * add IDEMPOTENCY_KEY_IGNORE_USER_FIELD to control if ignore user

[FTS-2464]

[FTS-2464]: https://hardcoretech.atlassian.net/browse/FTS-2464?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by xeonchen — APPROVED** [review]
LGTM

File: rest_framework_idempotency_key/middleware.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
 
 
 def get_current_user(request) -> Optional[settings.AUTH_USER_MODEL]:
+    if getattr(settings, 'IDEMPOTENCY_KEY_IGNORE_USER_FIELD', False):
+        return None
     if request.user:
```
Comment (kilikkuo) [inline rest_framework_idempotency_key/middleware.py:25]: Making this behavior depends on a setting causes it unchangeable during runtime.
Further, this change has impacts on all the APIs that rely on it.

As discussed with @xeonchen, let's remove `the dependency to request.user`.
That is, let's not care about whether the users between previous and current requests are the same or not.
And no more use of `def get_current_user`, and maybe just store a None into the DB user field.

**Review by kilikkuo — CHANGES_REQUESTED** [review]
It would be appreciated if you can help to revise the PR according to the comment.

File: rest_framework_idempotency_key/middleware.py (hunk 1 above)
Comment (leedan77) [reply rest_framework_idempotency_key/middleware.py]: Done

**Review by xeonchen — CHANGES_REQUESTED** [review]
You might want to remove the `user` field?

**Comment by leedan77** [comment]
> You might want to remove the `user` field?

Alright

File: rest_framework_idempotency_key/migrations/0003_remove_user.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Generated by Django 3.0.3 on 2022-04-13 07:54
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('rest_framework_idempotency', '0002_alter_idempotencykey_user'),
```
Comment (xeonchen) [inline rest_framework_idempotency_key/migrations/0003_remove_user.py:9]: 我在 repo 裡面沒有看到 `0002_alter_idempotencykey_user` 耶？可以檢查一下嗎？

Comment (leedan77) [reply rest_framework_idempotency_key/migrations/0003_remove_user.py]: Done

**Review by xeonchen — APPROVED** [review]

**Review by kilikkuo — APPROVED** [review]
Perfect !!! 
THANKS FOR YOUR EFFORT

**Pull request merged**